common/reviews/api/heft.api.md

```diff
@@ -248,8 +249,9 @@ export interface IHeftActionConfigurationOptions {
 }
 
 // @beta
-export interface IHeftBaseParameter {
+export interface IHeftBaseParameter<TValue> {
     readonly actionAssociated: boolean;
+    readonly value?: TValue;
     readonly valueProvided: boolean;
```

Comment:
@halfnibble  What if a parameter is omitted from the Heft command-line, but it has a default value, and therefore `IHeftBaseParameter.value` has a value?  Does that count as being `valueProvided`?  (🤔 I think so)

What if the parameter is a flag, and the flag was not omitted, and so the value is `false`.  Does that count as being `valueProvided`?  (🤔 now I'm not sure what is the point of `valueProvided`)

What if the parameter is a flag, and the flag was negated (something like `heft build --verbose --verbose=false`), and so the value is `false`.  Does that count as being `valueProvided`?  (🤔 now I'm not sure what is the point of `valueProvided`)

Practically speaking, is there a difference between `valueProvided` versus `IHeftBaseParameter.value !== undefined`?

It might be interesting for `ts-command-line` to expose some diagnostic details (Did the value come from the CLI or from an environment variable? Did they use the short or long form of the parameter name?), but I don't think program behavior should ever depend on these details.

Reply:
I myself was also questioning the purpose of `valueProvided`. I think all a plugin developer cares about is if the action called was associated with the parameter or not and if so, then what the value is.